types/react-redux/index.d.ts

```diff
@@ -46,7 +50,7 @@ interface ComponentMergeDecorator<TMergedProps, TOwnProps> {
  * @param mergeProps
  * @param options
  */
-export declare function connect(): ComponentDecorator<DispatchProp<any>, {}>;
+export declare function connect(): ComponentDecoratorInfer<DispatchProp<any>>;
```

Comment:
I see that only the zero argument overload for ```connect()``` uses ```ComponentDecoratorInfer``` now. Could you add a test that exercises it?